test/suite/test_checkpoint_snapshot03.py

```diff
@@ -0,0 +1,141 @@
+#   behaviour.
+
+def timestamp_str(t):
+    return '%x' % t
+class test_checkpoint_snapshot03(wttest.WiredTigerTestCase):
+
+    # Create a table.
+    uri = "table:test_checkpoint_snapshot03"
+    nrows = 500000
+
+    def conn_config(self):
+        config = 'cache_size=250MB,statistics=(all),statistics_log=(json,on_close,wait=1),log=(enabled=true)'
+        return config
+
+    def large_updates(self, uri, value, ds, nrows):
+        # Update a large number of records.
+        session = self.session
+        cursor = session.open_cursor(uri)
+        for i in range(0, nrows):
+            session.begin_transaction()
+            cursor[ds.key(i)] = value
+            session.commit_transaction()
+        cursor.close()
+
+    def check(self, check_value, uri, nrows):
+        session = self.session
+        session.begin_transaction()
+        cursor = session.open_cursor(uri)
+        count = 0
+        for k, v in cursor:
+            self.assertEqual(v, check_value)
+            count += 1
+        session.commit_transaction()
+        self.assertEqual(count, nrows)
+
+    def test_checkpoint_snapshot(self):
+
+        ds = SimpleDataSet(self, self.uri, 0, key_format="S", value_format="S",config='log=(enabled=false),leaf_page_max=4k')
+        ds.populate()
+        valuea = "aaaaa" * 100
+        valueb = "bbbbb" * 100
+        valuec = "ccccc" * 100
+
+        session1 = self.conn.open_session()
+        session1.begin_transaction()
+        cursor1 = session1.open_cursor(self.uri)
+        for i in range(self.nrows, self.nrows + 1):
+            cursor1.set_key(ds.key(i))
+            cursor1.set_value(valueb)
+            self.assertEqual(cursor1.insert(), 0)
+
+        self.large_updates(self.uri, valuea, ds, self.nrows)
+        self.check(valuea, self.uri, self.nrows)
+
+        self.session.checkpoint()
+        session1.rollback_transaction()
+        self.reopen_conn()
+
+        # Check the table contains the last checkpointed value.
+        self.check(valuea, self.uri, self.nrows)
+
+        session1 = self.conn.open_session()
+        session1.begin_transaction()
+        cursor1 = session1.open_cursor(self.uri)
+        for i in range(self.nrows, self.nrows + 1):
+            cursor1.set_key(ds.key(i))
+            cursor1.set_value(valueb)
+            self.assertEqual(cursor1.insert(), 0)
+
+        self.session.begin_transaction()
+        cursor = self.session.open_cursor(self.uri)
+        for i in range(0, 1):
+            cursor.set_key(ds.key(i))
+            cursor.set_value(valuec)
+            self.assertEqual(cursor.update(), 0)
+        self.session.commit_transaction()
+
+        self.session.checkpoint()
+        session1.rollback_transaction()
+
+        #Simulate a crash by copying to a new directory(RESTART1).
+        copy_wiredtiger_home(self, ".", "RESTART")
+
+        # Open the new directory.
+        self.conn = self.setUpConnectionOpen("RESTART")
```

Comment:
This might be intentional, so please disregard if so, but should the previous connection be closed before overwriting it?

There's a helper in `helper.py`- `simulate_crash_restart` that essentially implements lines 120-124 (with an additional connection close before restarting on a new connection).

Reply:
Thanks, Used the new helper function.